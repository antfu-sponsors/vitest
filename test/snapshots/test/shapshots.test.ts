const println = () => {
  const message = `
export default function () {
  function Foo() {
  }

  return Foo;
}
`
  return message
}

test('non default snapshot format', () => {
  expect({ foo: ['bar'] }).toMatchInlineSnapshot(`
  Object {
    "foo": Array [
      "bar",
    ],
  }
  `)
})

test('multiline strings ', () => {
  expect(println()).toMatchSnapshot()
})

test('updateInlineSnapshot should not remove end whitespace', () => {
  // issue #922
  expect(`
my string 
`).toMatchInlineSnapshot(`
  "
  my string 
  "
`)
})