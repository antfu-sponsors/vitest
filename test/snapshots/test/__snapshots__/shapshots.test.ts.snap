// Vitest Snapshot v1

exports[`js snapshots generated correctly 1`] = `
"import { describe, expect, test } from 'vitest'
import { validateNoSnapshots } from './validate-inline-js'

// when snapshots are generated Vitest reruns \`toMatchInlineSnapshot\` checks
// please, don't commit generated snapshots
describe('snapshots are generated in correct order', async () => {
  await validateNoSnapshots(__filename)

  test('first snaphot', () => {
    expect({ foo: ['bar'] }).toMatchInlineSnapshot(\`
      Object {
        \\"foo\\": Array [
          \\"bar\\",
        ],
      }
    \`)
  })

  test('second snapshot', () => {
    expect({ foo: ['zed'] }).toMatchInlineSnapshot(\`
      Object {
        \\"foo\\": Array [
          \\"zed\\",
        ],
      }
    \`)
  })
})
"
`;

exports[`multiline strings  1`] = `
"
export default function () {
  function Foo() {
  }

  return Foo;
}
"
`;
